packages/rum-core/src/domain/action/getActionNameFromElement.ts

```diff
@@ -1,26 +1,64 @@
 import { safeTruncate, isIE, find } from '@datadog/browser-core'
 import { getParentElement } from '../../browser/polyfills'
+import { NodePrivacyLevel, PRIVACY_ATTR_NAME } from '../../constants'
+import { getNodePrivacyLevel } from '../privacy'
 
 /**
  * Get the action name from the attribute 'data-dd-action-name' on the element or any of its parent.
  * It can also be retrieved from a user defined attribute.
  */
 export const DEFAULT_PROGRAMMATIC_ACTION_NAME_ATTRIBUTE = 'data-dd-action-name'
+export const ACTION_NAME_PLACEHOLDER = 'Masked Element'
+export function getActionNameFromElement(
+  element: Element,
+  privacyEnabledForActionName?: boolean,
+  userProgrammaticAttribute?: string
+): { name: string; masked?: boolean } {
```

Comment:
**Suggestion**: Both `privacyEnabledForActionName` and `userProgrammaticAttribute` are configuration values (or at least they should be, for clarity). To simplify a little bit, you could just take the `RumConfiguration` as a parameter:

```ts

export function getActionNameFromElement(
  element: Element,
  {
    enablePrivacyForActionName,
    actionNameAttribute: userProgrammaticAttribute
  }: RumConfiguration
): { name: string; masked?: boolean } {
```

Reply:
Nice suggestion. Yet, these parameters were optional. I think I could either pass configuration as optional and deconstruct in the function, or keep the deconstruction and make it required with type `RumConfiguration | Record<string, never>`. Maybe there are better ways to handle it? Wdyt

Reply:
These parameters are optional but in practice they are always provided, appart from the spec file. Let's require a `RumConfiguration`, and in the spec file we can fake it like `{} as RumConfiguration`